public/app/features/dashboard-scene/scene/NavToolbarActions.test.tsx

```diff
@@ -101,6 +102,24 @@ describe('NavToolbarActions', () => {
       expect(screen.queryByText(selectors.pages.Dashboard.DashNav.playlistControls.next)).not.toBeInTheDocument();
     });
   });
+
+  describe('Given new sharing button', () => {
+    it('Should show old share button when newDashboardSharingComponent FF is disabled', async () => {
+      setup();
+
+      expect(await screen.findByText('Share')).toBeInTheDocument();
+      const newShareButton = screen.queryByTestId(selectors.pages.Dashboard.DashNav.newShareButton.container);
+      expect(newShareButton).not.toBeInTheDocument();
+    });
+    it('Should show new share button when newDashboardSharingComponent FF is enabled', async () => {
+      config.featureToggles.newDashboardSharingComponent = true;
+      setup();
+
+      expect(screen.queryByTestId(selectors.pages.Dashboard.DashNav.shareButton)).not.toBeInTheDocument();
+      const newShareButton = await screen.findByTestId(selectors.pages.Dashboard.DashNav.newShareButton.container);
+      expect(newShareButton).toBeInTheDocument();
```

Comment:
- It's common to change the button text (and in this test, we're not testing that the button has "Share" text). Therefore, the test would fail. Actually, in the middle of the PR I changed "Share Link" to "Share" and the test failed when I was testing another thing.
- FindBy is async. However, the component should be already mounted. I can change it